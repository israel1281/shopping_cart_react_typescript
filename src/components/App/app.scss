.appContainer {
  .twoColumnGrid {
    display: grid;
    max-width: 1200px;
    margin: 50px auto auto;

    @media only screen and (min-width: 768px) {
      grid-template-columns: 1fr 4fr;
    }

    .side {
      display: grid;
      justify-content: center;
      padding: 15px;
      box-sizing: border-box;

      @media only screen and (min-width: 768px) {
        align-content: baseline;
      }
    }

    .main {
      .mainHeader {
        display: grid;
        grid-template-columns: 1fr 1fr;
        justify-content: end;
        padding: 0 15px;
        font-weight: 500;
      }
    }
  }
}
