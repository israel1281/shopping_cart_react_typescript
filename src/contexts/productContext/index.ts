export { ProductsProvider } from "./ProductContextProvider";
export { default as useProducts } from "./useProducts";
