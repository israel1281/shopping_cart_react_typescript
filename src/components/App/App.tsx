import { useEffect } from "react";

import { useProducts } from "../../contexts/productContext";
import "./app.scss";
import { useCart } from "../../contexts/cartContext";
import Loader from "../Loader";
import Filter from "../Filters/Filter";
import Products from "../Products";
import Cart from "../Cart";

function App() {
  const { isFetching, products, fetchProducts } = useProducts();
  const { closeCart } = useCart();

  useEffect(() => {
    fetchProducts();
  }, [fetchProducts]);
  return (
    <div className="appContainer">
      {isFetching && <Loader />}
      <div className="twoColumnGrid" onClick={closeCart}>
        <div className="side">
          <Filter />
        </div>
        <div className="main">
          <main className="mainHeader">
            <p>{products?.length} Product(s) found</p>
          </main>
          <Products products={products} />
        </div>
      </div>
      <Cart />
    </div>
  );
}

export default App;
