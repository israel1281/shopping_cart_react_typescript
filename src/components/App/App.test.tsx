import { render } from "@testing-library/react";
import { CartProvider } from "../../contexts/cartContext";
import { ProductsProvider } from "../../contexts/productContext";
import App from "./App";

describe("App component", () => {
  const setup = () => {
    return render(
      <ProductsProvider>
        <CartProvider>
          <App />
        </CartProvider>
      </ProductsProvider>
    );
  };

  test("should render correctly", () => {
    const view = setup();
    expect(view).toMatchSnapshot();
  });
});
