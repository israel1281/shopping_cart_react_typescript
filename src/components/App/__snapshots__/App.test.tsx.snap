// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="appContainer"
      >
        <div
          class="loaderContainer"
        >
          <div
            class="loader"
          />
          <div
            class="loader"
          />
          <div
            class="loader"
          />
          <div
            class="loader"
          />
        </div>
        <div
          class="twoColumnGrid"
        >
          <div
            class="side"
          >
            <div
              class="filterContainer"
            >
              <h4
                class="filterTitle"
              >
                Sizes:
              </h4>
              <div
                class="checkbox"
              >
                <label>
                  <input
                    data-testid="check_box"
                    type="checkbox"
                    value="XS"
                  />
                  <span
                    class="checkmark"
                    data-testid="check_label"
                  >
                    XS
                  </span>
                </label>
              </div>
              <div
                class="checkbox"
              >
                <label>
                  <input
                    data-testid="check_box"
                    type="checkbox"
                    value="S"
                  />
                  <span
                    class="checkmark"
                    data-testid="check_label"
                  >
                    S
                  </span>
                </label>
              </div>
              <div
                class="checkbox"
              >
                <label>
                  <input
                    data-testid="check_box"
                    type="checkbox"
                    value="M"
                  />
                  <span
                    class="checkmark"
                    data-testid="check_label"
                  >
                    M
                  </span>
                </label>
              </div>
              <div
                class="checkbox"
              >
                <label>
                  <input
                    data-testid="check_box"
                    type="checkbox"
                    value="ML"
                  />
                  <span
                    class="checkmark"
                    data-testid="check_label"
                  >
                    ML
                  </span>
                </label>
              </div>
              <div
                class="checkbox"
              >
                <label>
                  <input
                    data-testid="check_box"
                    type="checkbox"
                    value="L"
                  />
                  <span
                    class="checkmark"
                    data-testid="check_label"
                  >
                    L
                  </span>
                </label>
              </div>
              <div
                class="checkbox"
              >
                <label>
                  <input
                    data-testid="check_box"
                    type="checkbox"
                    value="XL"
                  />
                  <span
                    class="checkmark"
                    data-testid="check_label"
                  >
                    XL
                  </span>
                </label>
              </div>
              <div
                class="checkbox"
              >
                <label>
                  <input
                    data-testid="check_box"
                    type="checkbox"
                    value="XXL"
                  />
                  <span
                    class="checkmark"
                    data-testid="check_label"
                  >
                    XXL
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            class="main"
          >
            <main
              class="mainHeader"
            >
              <p>
                0
                 Product(s) found
              </p>
            </main>
            <div
              class="productsContainer"
            />
          </div>
        </div>
        <div
          class="cartContainer"
          style="right: -550px;"
        >
          <button
            class="cartButton"
            style="left: -50px; background-color: rgb(0, 44, 65); transition: all 0.3s;"
          >
            <div
              class="cartIcon"
            >
              <div
                class="cartQuantity"
              >
                0
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="appContainer"
    >
      <div
        class="loaderContainer"
      >
        <div
          class="loader"
        />
        <div
          class="loader"
        />
        <div
          class="loader"
        />
        <div
          class="loader"
        />
      </div>
      <div
        class="twoColumnGrid"
      >
        <div
          class="side"
        >
          <div
            class="filterContainer"
          >
            <h4
              class="filterTitle"
            >
              Sizes:
            </h4>
            <div
              class="checkbox"
            >
              <label>
                <input
                  data-testid="check_box"
                  type="checkbox"
                  value="XS"
                />
                <span
                  class="checkmark"
                  data-testid="check_label"
                >
                  XS
                </span>
              </label>
            </div>
            <div
              class="checkbox"
            >
              <label>
                <input
                  data-testid="check_box"
                  type="checkbox"
                  value="S"
                />
                <span
                  class="checkmark"
                  data-testid="check_label"
                >
                  S
                </span>
              </label>
            </div>
            <div
              class="checkbox"
            >
              <label>
                <input
                  data-testid="check_box"
                  type="checkbox"
                  value="M"
                />
                <span
                  class="checkmark"
                  data-testid="check_label"
                >
                  M
                </span>
              </label>
            </div>
            <div
              class="checkbox"
            >
              <label>
                <input
                  data-testid="check_box"
                  type="checkbox"
                  value="ML"
                />
                <span
                  class="checkmark"
                  data-testid="check_label"
                >
                  ML
                </span>
              </label>
            </div>
            <div
              class="checkbox"
            >
              <label>
                <input
                  data-testid="check_box"
                  type="checkbox"
                  value="L"
                />
                <span
                  class="checkmark"
                  data-testid="check_label"
                >
                  L
                </span>
              </label>
            </div>
            <div
              class="checkbox"
            >
              <label>
                <input
                  data-testid="check_box"
                  type="checkbox"
                  value="XL"
                />
                <span
                  class="checkmark"
                  data-testid="check_label"
                >
                  XL
                </span>
              </label>
            </div>
            <div
              class="checkbox"
            >
              <label>
                <input
                  data-testid="check_box"
                  type="checkbox"
                  value="XXL"
                />
                <span
                  class="checkmark"
                  data-testid="check_label"
                >
                  XXL
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          class="main"
        >
          <main
            class="mainHeader"
          >
            <p>
              0
               Product(s) found
            </p>
          </main>
          <div
            class="productsContainer"
          />
        </div>
      </div>
      <div
        class="cartContainer"
        style="right: -550px;"
      >
        <button
          class="cartButton"
          style="left: -50px; background-color: rgb(0, 44, 65); transition: all 0.3s;"
        >
          <div
            class="cartIcon"
          >
            <div
              class="cartQuantity"
            >
              0
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
